content/en/docs/reference/access-authn-authz/authentication.md

```diff
@@ -735,21 +735,32 @@ The following HTTP headers can be used to performing an impersonation request:
 * `Impersonate-User`: The username to act as.
 * `Impersonate-Group`: A group name to act as. Can be provided multiple times to set multiple groups. Optional. Requires "Impersonate-User"
 * `Impersonate-Extra-( extra name )`: A dynamic header used to associate extra fields with the user. Optional. Requires "Impersonate-User". In order to be preserved consistently, `( extra name )` should be lower-case, and any characters which aren't [legal in HTTP header labels](https://tools.ietf.org/html/rfc7230#section-3.2.6) MUST be utf8 and [percent-encoded](https://tools.ietf.org/html/rfc3986#section-2.1).
+* `Impersonate-Uid`: The {{< glossary_tooltip text="UID" term_id="uid" >}} associated with the user being impersonated. Optional. Requires "Impersonate-User".
```

Comment:
The tooltip reads:

> A Kubernetes systems-generated string to uniquely identify objects.

@sftim I find this confusing because you can impersonate any UID you want as long as you have the correct authorization.  This impersonated value is not Kubernetes generated and is not validated to be "correct" in any way.

Reply:
Ah OK. I wonder we don't we call these UUIDs (like Wikipedia, etc do)? UID is widely used in POSIX to mean user id (contrast GID).

Anyway, given that it's not linked to Kubernetes' `metadata.uid` field, I suggest amending this to either:
```suggestion
* `Impersonate-Uid`: A UUID that represents or is associated with the user being
   impersonated. Optional. Requires "Impersonate-User".
```
or
```suggestion
* `Impersonate-Uid`: An opaque string that represents or is associated with the user being
   impersonated. Optional. Requires "Impersonate-User". Kubernetes does not impose any
   format requirements on this string other than that it is non-empty.
```
Without reading the code, I can't guess which is meant and I imagine other readers might also have trouble working that out.

Reply:
Here are the [docs](https://github.com/kubernetes/kubernetes/blob/584658a17d958f0b663cb39dec1907cbba641300/staging/src/k8s.io/api/authentication/v1/types.go#L111-L115).  I am not sure if `A unique value that identifies this user across time.` is any more understandable for an end user though.

Reply:
How about:

```
An unique identifier that represents the user being impersonated. Optional. Requires "Impersonate-User". Kubernetes does not impose any format requirements on this string.
```

The "other than that it is non-empty" bit is weird since the API server would just ignore the header (there is no distinction between unset and empty).

Reply:
I'm still confused. The API doc suggests that UIDs aren't reused, which makes it sound like it's either a CustomResourceDefinition or some other Kubernetes API. On the other hand, if the UID is exogenous (like the example `42`) then who knows whether the other system that generates these UIDs reuses them or not.

I have seen (and, for that matter, am typing this on an example of) operating systems that will reuse userids.

If Kubernetes is _setting an expectation_ that UID reuse is problematic in the context of impersonation, should we document **that**?

Also, I can't imagine ever designing a system where a user can have a UID of [empty string] but who knows what weird other integration someone somewhere might try to connect? Every other string value is valid though.
If that part is too much detail, fine.

Reply:
I'm happy with
```suggestion
* A unique identifier that represents the user being impersonated. Optional. Requires
  "Impersonate-User". Kubernetes does not impose any format requirements on this string.
```
(not “an unique”, for semi-complicated reasons)